### DISCUSSION TITLE
Correct way to do span enrichment before .NET Core 5

### DISCUSSION THREAD

**Discussion by wolf8196** [RehanSaeed/Serilog.Enrichers.Span#7] [Q&A]
### Question

Hello.

From my understanding this package was created because System.Diagnostincs.Activity related information was made optional for .NET 5, but was included by default in lower versions.
Currently I would like to manually create an Activity and have TraceId, SpanId, etc included into consequent log entries.

Something like this:
```
using (new Activity("ActivityName").Start())
{
    // log entries with  TraceId, SpanId, Parent included
}
```

This code is outside of request pipeline, and .NET Core 3.1 logs are not enriched.
But if I migrate project to .NET 5 and add Span enricher, everything is included and works exactly as I need.

Could you please advise if there is a way to achieve the ".NET 5 + enricher" behaviour in .NET Core 3.1?
I assume the most straightforward way would be to manually "BeginScope" with all necessary information after the creation of Activity object, but maybe you are aware of better way.

Thank you

**Comment by RehanSaeed** [comment]
I'm not really sure. I vaguely remember reading that there is some sort of dependency on .NET 5 but I can't remember off hand.

Could you try debugging it and seeing what happens?

**Reply by wolf8196** [reply]
I tried to debug Serilogs internals with Source Link to see how it adds the Span info, but just couldn't make it work. It just refused to step into Serilogs code.

Dependency on .NET 5? Hmm. I only remember that they enabled W3C format by default...
